torchvision/prototype/datasets/_builtin/sintel.py

```diff
@@ -0,0 +1,182 @@
+        #     is not writeable, and PyTorch does not support non-writeable tensors."
+        return torch.from_numpy(np.copy(data_arr.reshape(h, w, 2).transpose(2, 0, 1)))
+
+    def _flows_key(self, data: Tuple[str, Any]) -> Tuple[str, int]:
+        path = pathlib.Path(data[0])
+        category = path.parent.name
+        idx = int(self._FILE_NAME_PATTERN.match(path.name).group("idx"))  # type: ignore[union-attr]
+        return category, idx
+
+    def _images_key(self, data: Tuple[Tuple[str, Any], Tuple[str, Any]]) -> Tuple[str, int]:
+        return self._flows_key(data[0])
+
+    def _collate_and_decode_sample(
+        self,
+        data: Tuple[Tuple[str, io.IOBase], Any],
+        *,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+        config: DatasetConfig,
+    ) -> Dict[str, Any]:
+        # flo, images = data
+        # img1, img2 = images
+
+        if config.split == "train":
+            flo, images = data
+            img1, img2 = images
+            flow_arr = self._read_flo(flo[1])
+        else:
+            # When split is `test`
+            img1, img2 = data
+
+        path1, buffer1 = img1
+        path2, buffer2 = img2
+
+        return dict(
+            image1=(path1, decoder(buffer1)) if decoder else (path1, buffer1),
+            image2=(path2, decoder(buffer2)) if decoder else (path2, buffer2),
+            flow=(flo[0], flow_arr) if config.split == "train" else ("", None),
+        )
+
+    def _make_datapipe(
+        self,
+        resource_dps: List[IterDataPipe],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> IterDataPipe[Dict[str, Any]]:
+        dp = resource_dps[0]
+        archive_dp = ZipArchiveReader(dp)
+
+        curr_split = Filter(archive_dp, self._filter_split, fn_kwargs=dict(split=config.split))
+        if config.split == "train":
+            flo_dp, pass_images_dp = Demultiplexer(
+                curr_split,
+                2,
+                partial(self._classify_archive, pass_name=config.pass_name),
+                drop_none=True,
+                buffer_size=INFINITE_BUFFER_SIZE,
+            )
+            flo_dp = Shuffler(flo_dp, buffer_size=INFINITE_BUFFER_SIZE)
+            pass_images_dp: IterDataPipe[Tuple[str, Any], Tuple[stry, Any]] = InSceneGrouper(pass_images_dp)
+            zipped_dp = IterKeyZipper(
+                flo_dp,
+                pass_images_dp,
+                key_fn=self._flows_key,
+                ref_key_fn=self._images_key,
+            )
+            return Mapper(zipped_dp, self._collate_and_decode_sample, fn_kwargs=dict(decoder=decoder, config=config))
+        else:
+            # When config.split is "test"
+            # There are no flow files for test split
+            # TODO: How to zip an empty IterDataPipe and pass to the Mapper?
+            # flo_dp = IterDataPipe()
```

Comment:
I'll need some help on this, @pmeier. I'm not sure how to create a `IterKeyZipper` from `flo_dp` (which will be `None`/empty here) and `pass_images_dp` (not `None`). Can you please take a look if you have the time?

Reply:
You don't need an `IterKeyZipper` here, because, like you said, `flo_dp` is empty or better non-existent. To add fake flow data, you can simply do

```python
pass_images_dp = Mapper(pass_images_dp, lambda data: ((None, None), data))
```

Now `pass_images_dp` and `zipped_dp` from the train split should have the same format and can easily be processed by `_collate_and_decode_sample`.

Reply:
Awesome, thanks Philip! This should be fixed now. :)